network/network_windows.go

```diff
@@ -318,6 +325,47 @@ func (nm *networkManager) configureHcnNetwork(nwInfo *NetworkInfo, extIf *extern
 	return hcnNetwork, nil
 }
 
+func (nm *networkManager) addIPv6DefaultRoute() error {
+	// add ipv6 default route if it does not exist in dualstack overlay windows node from persistentstore
+	cmd := `((Get-NetIPInterface | where InterfaceAlias  -Like "vEthernet*").IfIndex)[0]`
+	ifIndex, err := nm.plClient.ExecutePowershellCommand(cmd)
+	if err != nil {
+		return errors.Wrapf(err, "error while executing powershell command to get ipv6 Hyper-V interface")
+	}
+
+	getIPv6RouteActiveCmd := fmt.Sprintf("Get-NetRoute -DestinationPrefix %s", defaultIPv6Route)
+	getIPv6RoutePersistentCmd := fmt.Sprintf("Get-NetRoute -DestinationPrefix %s -PolicyStore Persistentstore", defaultIPv6Route)
+	if out, err := nm.plClient.ExecutePowershellCommand(getIPv6RoutePersistentCmd); err != nil {
+		logger.Info("ipv6 default route is not found from persistentstore, adding default ipv6 route to the windows node", zap.Any("out", out))
+		// run powershell cmd to add ipv6 default route
+		addCmd := fmt.Sprintf("Remove-NetRoute -DestinationPrefix %s -InterfaceIndex %s -NextHop %s -confirm:$false;New-NetRoute -DestinationPrefix %s -InterfaceIndex %s -NextHop %s -confirm:$false",
+			defaultIPv6Route, ifIndex, defaultIPv6NextHop, defaultIPv6Route, ifIndex, defaultIPv6NextHop)
+
+		// if command is failed to execute, then attempt to execute extra two times
+		for i := 0; i <= addIPv6DefaultRouteAttempts; i++ {
+			if out, err := nm.plClient.ExecutePowershellCommand(addCmd); err != nil {
+				logger.Error("Failed to add ipv6 default gateway route, retrying after error", zap.Any("out", out), zap.Error(err))
+			} else {
+				// check if ipv6 default route is added to both active and persistent store
+				if out, err := nm.plClient.ExecutePowershellCommand(getIPv6RoutePersistentCmd); err != nil {
+					logger.Error("default Route is not added to windows persistent store", zap.Any("out", out))
+					return errors.Wrapf(err, "Failed to add ipv6 default gateway route")
```

Comment:
can it be `Failed to get ipv6 default gateway route in persistent store`. Also update logger.Error with same message?